### PR TITLE
feat: add ability to reset libem.calibrate calls

### PR DESCRIPTION
**Pull request by daiwaid**
#### What this PR does / why we need it:
- add ability to reset libem.calibrate calls (necessary for libem serve functionality)

#### Which issue(s) this PR addresses (if any):

Addresses #

#### Does this PR introduce a user-facing change?

```release-note
- libem.reset() no longer resets external model clients, use libem.model.reset() instead
```

### CODE REVIEW

File: libem/tune/calibrate/function.py [hunk 1]
```diff
@@ -24,13 +26,22 @@ def func(config, verbose=False):
         except:
             raise Exception(f"unable to find parameter {path}")
         assert isinstance(parameter, Parameter), f"invalid parameter {path}"
+        
+        # Cache the original value
+        if path not in cache:
+            cache[path] = parameter()
```
Comment (zenodflow) [inline libem/tune/calibrate/function.py:32]: Liked this simple change! But was wondering if we should use the parameter struct itself to store cache value vs. this "external" cache.

Comment (zenodflow) [reply libem/tune/calibrate/function.py:32]: I prefer the current design - but let's have a think over the alternative?

Comment (daiwaid) [reply libem/tune/calibrate/function.py:32]: I think it depends on what we define as "reset". Currently the design is to reset `libem.calibrate()` calls (modifications directly from the user), so other means of modifying parameters (i.e. optimizing via grid search, loading from a config file, etc) will not be reset. If we want to strictly reset all parameters to their default values then moving this to the Parameter class may be better.

Comment (zenodflow) [reply libem/tune/calibrate/function.py:32]: Got it. That makes sense!

File: libem/tune/calibrate/function.py [hunk 2]
```diff
@@ -24,13 +26,22 @@ def func(config, verbose=False):
         except:
             raise Exception(f"unable to find parameter {path}")
         assert isinstance(parameter, Parameter), f"invalid parameter {path}"
+        
+        # Cache the original value
+        if path not in cache:
```
Comment (zenodflow) [inline libem/tune/calibrate/function.py:31]: Does this support multiple reset calls? Or just reset to the first / original value?

File: libem/tune/calibrate/function.py [hunk 3]
```diff
@@ -10,6 +10,8 @@
 schema = {}
 
 
+cache = {}
```
Comment (zenodflow) [inline libem/tune/calibrate/function.py:13]: Could we add a brief explanation of how the cache works?

File: libem/tune/calibrate/function.py (hunk 2 above)
Comment (daiwaid) [reply libem/tune/calibrate/function.py:31]: It resets to the original (non-calibrated) value.

File: libem/tune/calibrate/function.py (hunk 3 above)
Comment (daiwaid) [reply libem/tune/calibrate/function.py:14]: Done. It only saves the parameter value the first time said parameter is being calibrated so it only saves the original/non-calibrated value.

**Pull request merged**